core/src/main/java/org/axonframework/config/EventProcessingConfigurer.java

```diff
@@ -0,0 +1,444 @@
+     * @param interceptorBuilder The builder function that provides an interceptor for each available processor
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerDefaultHandlerInterceptor(
+            BiFunction<Configuration, String, MessageHandlerInterceptor<? super EventMessage<?>>> interceptorBuilder);
+
+    /**
+     * Registers the sequencing policy created by given {@code policyBuilder} to the processing group with given
+     * {@code processingGroup}. Any previously configured policy for the same name will be overwritten.
+     *
+     * @param processingGroup The name of the processing group to assign the sequencing policy for
+     * @param policyBuilder   The builder function to create the policy to use
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerSequencingPolicy(String processingGroup,
+                                                       Function<Configuration, SequencingPolicy<? super EventMessage<?>>> policyBuilder);
+
+    /**
+     * Registers the sequencing policy created by given {@code policyBuilder} to the processing groups for which no
+     * explicit policy is defined (using {@link #registerSequencingPolicy(String, Function)}).
+     * <p>
+     * Defaults to a {@link SequentialPerAggregatePolicy}.
+     *
+     * @param policyBuilder The builder function to create the policy to use
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerDefaultSequencingPolicy(
+            Function<Configuration, SequencingPolicy<? super EventMessage<?>>> policyBuilder);
+
+    /**
+     * Registers the builder function to create the Message Monitor for the {@link EventProcessor} of the given name.
+     *
+     * @param name                  The name of the event processor
+     * @param messageMonitorBuilder The builder function to use
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    default EventProcessingConfigurer registerMessageMonitor(String name,
+                                                             Function<Configuration, MessageMonitor<Message<?>>> messageMonitorBuilder) {
+        return registerMessageMonitorFactory(name,
+                                             (configuration, componentType, componentName) -> messageMonitorBuilder
+                                                     .apply(configuration));
+    }
+
+    /**
+     * Registers the factory to create the Message Monitor for the {@link EventProcessor} of the given name.
+     *
+     * @param name                  The name of the event processor
+     * @param messageMonitorFactory The factory to use
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerMessageMonitorFactory(String name,
```

Comment:
rename first parameter to `eventProcessorName`? This would make it more obvious as to what name is required